src/features/movies/moviesSaga.js

```diff
@@ -1,12 +1,14 @@
 import { takeLatest, call, put, delay } from "redux-saga/effects";
-import { getPopularMovies, getGenres, getMovie, getMovieCredits } from "../../../src/apiClient";
-import { fetchPopularMovies, fetchPopularMoviesSucces, fetchPopularMoviesError, setGenres, fetchMovieSucces, fetchMovieError, fetchMovie } from "./moviesSlice";
+import { getPopularMovies, getGenres, getMovie, getMovieCredits, searchForMovies } from "../../../src/apiClient";
+import { fetchPopularMovies, fetchPopularMoviesSucces, fetchPopularMoviesError, setGenres, fetchMovieSucces, fetchMovieError, fetchMovie, resetMovies } from "./moviesSlice";
 
 function* fetchPopularMoviesHandler({ payload }) {
     try {
+        yield put(resetMovies());
         yield delay(500);
```

Comment:
Usunąłbym te delaye, przedtem były dobre aby widzieć jak komponent ładowania się zachowuje ale teraz wydają mi się zbędne. Jedyne opóźnienie celowe jakie powinno być to to związane z rozpoczęciem wyszukiwania :) 
A pozostałe tematy powinny działać jak najszybciej :)